Fix account history contract in the simple bank account application

KodaiD opened the pull request:
## Description

There is a bug in the account-history contract in the simple bank account application, and it does not display the asset data as follows:

```
❯ ../scalardl-java-client-sdk/client/bin/scalardl execute-contract --properties ../conf/client.properties --contract-id account-history --contract-argument '{"id": "a111"}'
Contract result:
{
  "status" : "succeeded",
  "history" : [ {
    "id" : "a111",
    "data" : "",
    "age" : 1
  }, {
    "id" : "a111",
    "data" : "",
    "age" : 0
  } ]
}
```

When I updated the application in #64, I overlooked this bug because the document doesn't give any instructions on how to use the account-history. So I've also added the instruction for running the account-history to the document in addition to the bug fix.

## Related issues and/or PRs

- https://github.com/scalar-labs/scalardl/pull/61

## Changes made

- Fix the bug in the account-history contract
- Update the document to describe how to use the account-history contract

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

## Release notes

N/A

KodaiD commented:
I've confirmed that all steps related to contract executions have been executed successfully.

```
applications/simple-bank-account/contract on  fix-account-history-contract-in-sample-app [!?] via 🅶 via ☕ v21.0.4
❯ ${SCALAR_SDK_HOME}/client/bin/scalardl execute-contract --properties ../conf/client.properties --contract-id create-account --contract-argument '{"id": "a111"}'
Contract result:
{
  "status" : "succeeded",
  "message" : "account a111 created"
}

applications/simple-bank-account/contract on  fix-account-history-contract-in-sample-app [!?] via 🅶 via ☕ v21.0.4 took 6s
❯ ${SCALAR_SDK_HOME}/client/bin/scalardl execute-contract --properties ../conf/client.properties --contract-id create-account --contract-argument '{"id": "b222"}'
Contract result:
{
  "status" : "succeeded",
  "message" : "account b222 created"
}

applications/simple-bank-account/contract on  fix-account-history-contract-in-sample-app [!?] via 🅶 via ☕ v21.0.4 took 6s
❯ ${SCALAR_SDK_HOME}/client/bin/scalardl execute-contract --properties ../conf/client.properties --contract-id deposit --contract-argument '{"id": "a111", "amount": 100}'
Contract result:
{
  "status" : "succeeded",
  "old_balance" : 0,
  "new_balance" : 100
}

applications/simple-bank-account/contract on  fix-account-history-contract-in-sample-app [!?] via 🅶 via ☕ v21.0.4 took 6s
❯ ${SCALAR_SDK_HOME}/client/bin/scalardl execute-contract --properties ../conf/client.properties --contract-id transfer --contract-argument '{"from": "a111", "to": "b222", "amount": 100}'
Contract result:
{
  "status" : "succeeded",
  "from_old_balance" : 100,
  "from_new_balance" : 0,
  "to_old_balance" : 0,
  "to_new_balance" : 100
}

applications/simple-bank-account/contract on  fix-account-history-contract-in-sample-app [!?] via 🅶 via ☕ v21.0.4 took 6s
❯ ${SCALAR_SDK_HOME}/client/bin/scalardl execute-contract --properties ../conf/client.properties --contract-id account-history --contract-argument '{"id": "a111"}'
Contract result:
{
  "status" : "succeeded",
  "history" : [ {
    "id" : "a111",
    "age" : 2,
    "data" : {
      "balance" : 0
    }
  }, {
    "id" : "a111",
    "age" : 1,
    "data" : {
      "balance" : 100
    }
  }, {
    "id" : "a111",
    "age" : 0,
    "data" : {
      "balance" : 0
    }
  } ]
}

jnmt reviewed (changes requested):
Thank you for the update! I left one request. Can you handle it when you have time?

kota2and3kan commented on docs/applications/simple-bank-account/contract/src/main/java/com/scalar/application/bankaccount/contract/AccountHistory.java:
```diff
@@ -41,8 +41,8 @@ public JsonNode invoke(Ledger<JsonNode> ledger, JsonNode argument, @Nullable Jso
                   .put("id", asset.id())
-                  .put("data", asset.data().asText())
-                  .put("age", asset.age());
+                  .put("age", asset.age())
+                  .set("data", asset.data());
```
FYI (For reviewers):

In this sample application, the `balance` data is stored as an asset data as follows.

```sql
postgres=# select id, age, output from scalar.asset;
  id  | age |     output
------+-----+-----------------
 a111 |   0 | {"balance":0}
 b222 |   0 | {"balance":0}
 a111 |   1 | {"balance":100}
 a111 |   2 | {"balance":0}
 b222 |   1 | {"balance":100}
(5 rows)
```

And, `asset.data()` returns `com.fasterxml.jackson.databind.JsonNode` when we use Jackson.
https://javadoc.io/doc/com.scalar-labs/scalardl-java-client-sdk/latest/index.html

In this case, it seems that we should use `set(String propertyName, JsonNode value)` to set `JsonNode` as value instead of using `put(String fieldName, String v)`. 

- put(String fieldName, String v)
  - https://javadoc.io/static/com.fasterxml.jackson.core/jackson-databind/2.17.0/com/fasterxml/jackson/databind/node/ObjectNode.html#put-java.lang.String-java.lang.String-
- set(String propertyName, JsonNode value)
  - https://javadoc.io/static/com.fasterxml.jackson.core/jackson-databind/2.17.0/com/fasterxml/jackson/databind/node/ObjectNode.html#set-java.lang.String-com.fasterxml.jackson.databind.JsonNode-

(I'm not familiar with `Jackson`, but according to the test result, it seems that the `put(String fieldName, String v)` method does not show expected result even if we set `JsonNode` as String data by using `.asText()` method.)

Therefore, to set `JsonNode` in the contract result (i.e., to set `balance` data in `ArrayNode result`), it seems that we have to use `set(String propertyName, JsonNode value)`.

jnmt replied:
> (I'm not familiar with Jackson, but according to the test result, it seems that the put(String fieldName, String v) method does not show expected result even if we set JsonNode as String data by using .asText() method.)

@kota2and3kan 
Thank you for testing this. I think it's expected because `asText()`

> will return a valid String representation of the container value, **if the node is a value node** (method [JsonNode.isValueNode()](https://javadoc.io/static/com.fasterxml.jackson.core/jackson-databind/2.17.0/com/fasterxml/jackson/databind/JsonNode.html#isValueNode--) returns true), **otherwise empty String**.

https://javadoc.io/static/com.fasterxml.jackson.core/jackson-databind/2.17.0/com/fasterxml/jackson/databind/node/ContainerNode.html#asText--

So, we should not use it to stringify an `ObjectNode` or `ArrayNode`.

kota2and3kan replied:
@jnmt 
Ah, I see. I didn't look into `asText()`. Thank you for the information! 🙇‍♂️

kota2and3kan reviewed (approved):
LGTM! Thank you!

FYI: I was able to see the expected result as follows in my local environment.

```json
$ ./scalardl-java-client-sdk-3.9.1/bin/scalardl execute-contract --config ./conf/client.properties --contract-id account-history --contract-argument '{"id": "a111"}'
Contract result:
{
  "status" : "succeeded",
  "history" : [ {
    "id" : "a111",
    "age" : 2,
    "data" : {
      "balance" : 0
    }
  }, {
    "id" : "a111",
    "age" : 1,
    "data" : {
      "balance" : 100
    }
  }, {
    "id" : "a111",
    "age" : 0,
    "data" : {
      "balance" : 0
    }
  } ]
}
```

kota2and3kan commented:
@jnmt 

> Thank you for the update! I left one request. Can you handle it when you have time?

BTW, I cannot see your request. Could you please check if your comment applied properly on GitHub?

jnmt commented:
@KodaiD Sorry, I probably made a mistake. My request was related to the document part. Since we have moved the documents to the `docs-internal-scalardl`, can you update it in the `docs-internal-scalardl` instead of this repository to reflect the changes properly?

KodaiD commented:
@jnmt Thanks for reviewing! I've removed the update in README.md in this repo. I've updated the documentation in doc-internal (in https://github.com/scalar-labs/docs-internal-scalardl/pull/226).

feeblefakie reviewed (approved):
LGTM! Thank you!
